addons/event_sale/models/sale_order.py

```diff
@@ -136,6 +136,9 @@ def _compute_price_unit(self):
             if not line.product_id or line._origin.product_id != line.product_id:
                 # Note: this also computes the price for new lines :)
                 super(SaleOrderLine, line)._compute_price_unit()
+            if line.product_id and line._origin.product_id == line.product_id:
+                # Note: compute the price whenever possible from the ticket itself
+                line.price_unit = line.event_ticket_id.price
```

Comment:
If we want to always recompute the Sales Order line price, then you should simply remove the whole `_compute_price_unit` override, the correct price computation is specified in the super method, and correctly considers 1) the pricelist discount policy to include the discount in the price or not 2) the included taxes.

Reply:
after discussion modified and it works :)